src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java

```diff
@@ -54,11 +56,19 @@ public ProviderSqlSource(Configuration configuration, Object provider) {
    * @since 3.4.5
    */
   public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod) {
+    this(configuration, provider, mapperType, mapperMethod, new XMLLanguageDriver());
```

Comment:
I think better using the `Configuration#getDefaultScriptingLanguageInstance()` instead of `new XMLLanguageDriver()`.